package/src/components/ChannelList/hooks/listeners/useNewMessage.ts

```diff
@@ -34,27 +35,22 @@ export const useNewMessage = <
       } else {
         setChannels((channels) => {
           if (!channels) return channels;
+          const channelInList = channels.filter((channel) => channel.cid === event.cid).length > 0;
```

Comment:
why move this before the if though? 

if lock was enabled.. we wouldnt need to filter

Reply:
or was that behaviour a bug?

Reply:
No, it wasn't a bug. I have just aligned it the same as React SDK, and now it looks better and cleaner, IMO.